docs/source/configuration.rst

```diff
@@ -374,6 +374,19 @@ to use them for templated. In the above example, you might define a file at
         return "GROUP BY 1,2"
 
 
+If you need root-level functions, you can define them in __init__.py.
+
+.. code-block:: jinja
+
+    SELECT {{ custom_sum('foo', 'bar') }} FROM baz
+
+`sqlfluff_libs/__init__.py`:
+
+.. code-block:: python
+
+    def custom_sum(a: str, b: str) -> str:
+        return a + b
+
```

Comment:
There's a bit in this section of the docs that says
> This will pull in any python modules from that directory and allow sqlfluff to use them for templated.

Maybe just add a slight caveat to reflect what we discussed about this not importing modules from subfolders of the supplied path:
e.g.
 > This will pull in any python modules from that directory (excluding subdirectories) and allow sqlfluff to use them in templates.

Think even if we stick with `__init__.py` as our module for root functions these changes should at least make the behaviour clear to people reading the docs 👍